cfme/tests/infrastructure/test_host.py

```diff
@@ -617,3 +617,22 @@ def test_infrastructure_hosts_crud(appliance, setup_provider):
     # Case6 - lastly do the delete. First try is canceled.
     host.delete(cancel=True)
     host.delete
+
+
+@test_requirements.infra_hosts
+def test_infrastructure_hosts_tagging(appliance, setup_provider):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: high
+        initialEstimate: 1/6h
+    """
+    host = appliance.collections.hosts.all()[0]
+    tag = host.add_tag()
```

Comment:
Required: Please add teardown for it. so added tag can remove after test.

Reply:
Good catch, thanks. I knew I was using existing tags so I didn't need to remove the actual tag, but forgot I'd need to remove the tag association from the host. Done. I didn't put the cleanup in a context manager or try/except because the only think that could fail and stop the cleanup would be the thing we are cleaning up (addition of tag).